common/src/main/java/com/wynntils/models/characterstate/CharacterStateModel.java

```diff
@@ -33,8 +35,11 @@ public final class CharacterStateModel extends Model {
     private final PowderSpecialSegment powderSpecialSegment = new PowderSpecialSegment();
     private final SprintSegment sprintSegment = new SprintSegment();
 
-    private static final String WYNN_DEATH_MESSAGE = "§r §4§lYou have died...";
-    private Vec3 lastLocation = new Vec3(0, 0, 0);
+    // we need a .* in front because the message may have a custom timestamp prefix (or some other mod could do
+    // something weird)
+    private static final Pattern WYNN_DEATH_MESSAGE = Pattern.compile(".*§r §4§lYou have died\\.\\.\\.");
+    private Vec3 lastLocationBeforeTeleport;
```

Comment:
```suggestion
    private Position lastPositionBeforeTeleport;
```

Reply:
I'd say no, a better design principle is actually the exact opposite: never declare a field (or a method argument) as a more concrete type than you have to. That is, better to call something `List<String>` than `LinkedList<String>`. Then you can still instantiate it as a LinkedList, but you do not give any guarantees that it is anything but a List. The only reason you would ever need to declare it as a LinkedList if you needed to access methods that are only available on LinkedList, like `removeFirst()` or whater it is called.

Reply:
Ah alright